tests/integrationv2/test_session_resumption.py

```diff
@@ -102,3 +100,218 @@ def test_session_resumption_s2n_client(managed_process, cipher, curve, protocol,
         assert results.exception is None
         assert results.exit_code == 0
         assert results.stdout.count(bytes("6 server accepts that finished".encode('utf-8')))
+
+
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("cipher", TLS13_CIPHERS, ids=get_parameter_name)
+@pytest.mark.parametrize("curve", ALL_TEST_CURVES, ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", ALL_TEST_CERTS, ids=get_parameter_name)
+@pytest.mark.parametrize("protocol", [Protocols.TLS13], ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [OpenSSL], ids=get_parameter_name)
+def test_tls13_session_resumption_s2n_server(managed_process, tmp_path, cipher, curve, protocol, provider, certificate):
+    port = str(next(available_ports))
+
+    # Use temp directory to store session tickets
+    p = tmp_path / 'ticket.pem'
+    path_to_ticket = str(p)
+
+    client_options = ProviderOptions(
+        mode=Provider.ClientMode,
+        host="localhost",
+        port=port,
+        cipher=cipher,
+        curve=curve,
+        insecure=True,
+        reconnect=False,
+        data_to_send=data_bytes(4069),
+        extra_flags = ['-sess_out', path_to_ticket],
+        protocol=protocol)
+
+    server_options = copy.copy(client_options)
+    server_options.mode = Provider.ServerMode
+    server_options.key = certificate.key
+    server_options.cert = certificate.cert
+    server_options.use_session_ticket = True
+    server_options.extra_flags = None
+
+    server = managed_process(S2N, server_options, timeout=5)
+    client = managed_process(provider, client_options, timeout=5)
+
+    # The client should have received a session ticket
+    for results in client.get_results():
+        assert results.exception is None
+        assert results.exit_code == 0
+        assert b'Post-Handshake New Session Ticket arrived:' in results.stdout
+
```

Comment:
Not a blocker because this isn't the only test we've written this way, but:
What's the point of parameterizing the test to take a "provider" if we then check the provider's output for messages that only OpenSSL prints? Shouldn't we focus on using S2N for verification? Or if the test is written to only work for OpenSSL, can we just be honest about that and not use the provider parameter?

Reply:
Okay I tried to do this and it turns out the test errors if you don't use the provider parameter 🤷‍♀️